autonomica-api/app/services/keyword_clustering.py

```
@@ -0,0 +1,741 @@
+                    
+                    # Analyze primary keyword
+                    cluster.primary_keyword = cluster.keywords[0]
+                    
+                    # Analyze intent
+                    intent_analysis = self.analyze_keyword_intent(cluster.primary_keyword)
+                    cluster.intent_type = intent_analysis.intent_type
+                    
+                    # Calculate cluster score based on size and cohesion
+                    if cluster.size > 1:
+                        cluster_similarities = []
+                        for i, kw1 in enumerate(cluster.keywords):
```

Ruff (0.12.2)

Loop control variable `kw1` not used within loop body

Rename unused `kw1` to `_kw1`

(B007)